eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/ImpTest.hs

```diff
@@ -329,22 +331,23 @@ trySubmitVote ::
         [PredicateFailure (EraRule "LEDGER" era)]
         (TxId (EraCrypto era))
     )
-trySubmitVote vote voter gaId = do
-  trySubmitTx $
-    mkBasicTx mkBasicTxBody
-      & bodyTxL . votingProceduresTxBodyL
-        .~ VotingProcedures
-          ( Map.singleton
-              voter
-              ( Map.singleton
-                  gaId
-                  ( VotingProcedure
-                      { vProcVote = vote
-                      , vProcAnchor = SNothing
-                      }
-                  )
-              )
-          )
+trySubmitVote vote voter gaId =
+  fmap (fmap txIdTx) $
```

Comment:
We wouldn't have to do this if `trySubmitTx` returned a `TxId`, see my comment above

Reply:
See my other comment.

If it proves to be a common enough pattern then we can create extra helper functions that return TxId instead of Tx, but calling `txIdTx` on the result is quite easy.